test/autorest/additionalpropsgroup/zz_generated_models.go

```diff
@@ -56,23 +55,15 @@ type CatAPTrueResponse struct {
 }
 
 type Error struct {
+	raw     string
 	Message *string `json:"message,omitempty"`
 	Status  *int32  `json:"status,omitempty"`
 }
 
 // Error implements the error interface for type Error.
+// The contents of the error text are not contractual and subject to change.
 func (e Error) Error() string {
-	msg := ""
-	if e.Message != nil {
-		msg += fmt.Sprintf("Message: %v\n", *e.Message)
-	}
-	if e.Status != nil {
-		msg += fmt.Sprintf("Status: %v\n", *e.Status)
-	}
-	if msg == "" {
-		msg = "missing error info"
-	}
-	return msg
+	return e.raw
```

Comment:
Wondering whether we'd need to ever let customers know that they could potentially access the error fields on the error type? Should we differentiate between when we fail to unmarshal and not?

Reply:
Just because it's a bit of a waste of unmarshalling if nobody ever uses the fields....why would we keep unmarshalling then? Thinking out loud here...

Reply:
You bring up good points.  We need to update the doc comments indicating the error type(s) returned from the APIs so that callers can get to the concrete types.

I think the usefulness of the concrete error types is going to be service and context dependent (e.g. blob storage seems to care about this).

Reply:
I agree, to clarify that the structured info wont be lost. Let me know if you want to add that in this PR.

Reply:
Done.